synse/http_integration_test.go

```diff
@@ -0,0 +1,551 @@
+		assert.Equal(t, 2, len(plugin.Health.Checks))
+
+		for _, check := range plugin.Health.Checks {
+			assert.NotEmpty(t, check.Name)
+			assert.Equal(t, "OK", check.Status)
+			assert.NotEmpty(t, check.Type)
+			assert.NotEmpty(t, check.Timestamp)
+			assert.Empty(t, check.Message)
+		}
+
+	}
+}
+
+func TestIntegration_PluginHealth(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	health, err := client.PluginHealth()
+	assert.NoError(t, err)
+	assert.NotEmpty(t, health.Status)
+	assert.NotEmpty(t, health.Updated)
+	assert.Equal(t, 1, len(health.Healthy))
+	assert.Equal(t, 0, len(health.Unhealthy))
+	assert.Equal(t, 1, health.Active)
+	assert.Equal(t, 0, health.Inactive)
+}
+
+func TestIntegration_Scan(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	opts := scheme.ScanOptions{}
+	devices, err := client.Scan(opts)
+	assert.NoError(t, err)
+	assert.Equal(t, 22, len(devices))
+
+	for _, device := range devices {
+		assert.NotEmpty(t, device.ID)
+		assert.NotEmpty(t, device.Info)
+		assert.NotEmpty(t, device.Type)
+		assert.NotEmpty(t, device.Plugin)
+		assert.NotEmpty(t, device.Tags)
+
+		// only led devices have alias
+		if device.Type == "led" {
+			assert.NotEmpty(t, device.Alias)
+		} else {
+			assert.Empty(t, device.Alias)
+		}
+	}
+}
+
+func TestIntegration_Tags(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	opts := scheme.TagsOptions{}
+	tags, err := client.Tags(opts)
+	assert.NoError(t, err)
+	assert.Equal(t, 10, len(tags))
+
+	for _, tag := range tags {
+		assert.NotEmpty(t, tag)
+	}
+}
+
+func TestIntegration_Info(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	opts := scheme.ScanOptions{}
+	devices, err := client.Scan(opts)
+	assert.NoError(t, err)
+	assert.Equal(t, 22, len(devices))
+
+	for _, device := range devices {
+		assert.NotEmpty(t, device)
+	}
+
+	for _, device := range devices {
+		assert.NotEmpty(t, device)
+
+		info, err := client.Info(device.ID)
+		assert.NoError(t, err)
+		assert.NotEmpty(t, info.Timestamp)
+		assert.NotEmpty(t, info.ID)
+		assert.NotEmpty(t, info.Type)
+		assert.NotEmpty(t, info.Plugin)
+		assert.NotEmpty(t, info.Capabilities.Mode)
+		assert.NotEmpty(t, info.Tags)
+		assert.NotEmpty(t, info.Metadata)
+		assert.Empty(t, info.Capabilities.Write.Actions)
+
+		// TODO - add sort_index to the scheme
+
+		// only led devices have alias
```

Comment:
see comment above, I'd remove this unless we use a custom emulator device config.